api/v3/Order.php

```diff
@@ -127,6 +127,40 @@ function civicrm_api3_order_create(&$params) {
 }
 
 /**
+ * Cancel a Order.
+ *
+ * @param array $params
+ *   Input parameters.
+ *
+ * @return array
+ */
+function civicrm_api3_order_cancel($params) {
+  $contributionStatuses = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
+  $params['contribution_status_id'] = array_search('Cancelled', $contributionStatuses);
+  $result = civicrm_api3('Contribution', 'create', $params);
+  if (!CRM_Utils_Array::value('is_error', $result)) {
```

Comment:
This 'IF' is not required - the api will throw an exception if you call it using the civicrm_api3 wrapper (it won't if you call it with civicrm_api() wrapper).

What that means is you can leave out the IF clause because if it gets that far it has succeeded. You don't need to catch / handle the exception unless you have something to add to it.